app/mm-redux/actions/posts.ts

```diff
@@ -214,6 +215,29 @@ export function createPost(post: Post, files: any[] = []) {
             },
         });
 
+        const {message} = post;
+        const RE_UNICODE_EMOJI = emojiRegex();
+        const emojis = message.match(RE_UNICODE_EMOJI);
+        function emojiUnicode(input:string) {
+            if (input.length === 1) {
+                return input.codePointAt(0)?.toString(16);
+            }
+            return input.codePointAt(0)?.toString(16) + '-' + input.codePointAt(1)?.toString(16);
+        }
+        if (emojis) {
+            const emojisAvailableWithMatterMost = [];
+            for (const emoji of emojis) {
+                const unicode = emojiUnicode(emoji);
+                const index = EmojiIndicesByUnicode.get(unicode || '');
+                if (index) {
+                    emojisAvailableWithMatterMost.push(Emojis[index].aliases[0]);
+                }
+            }
+            dispatch({
+                type: ViewTypes.ADD_RECENT_EMOJI_ARRAY,
+                emojis: emojisAvailableWithMatterMost,
+            });
+        }
```

Comment:
this is a step in  the right direction but again a few things:
1. This should be its own action
2. The action should also consider custom emojis (non-unicode)
3. Action should be called from the component after posting the message has been successful

Reply:
@enahum So the action 
```js
{ type: ADD_REACTIONS_IN_TEXT_TO_RECENT_REACTIONS, text: text}
 ```
right?

Reply:
@enahum How could we know inside the component if the posting is successful? Do we need to return the result of `Client4.createPost` from the `createPost` function ?

Reply:
I mean an action creator, like an independent function (maybe call it `recentUsedEmojisInMessage`). under `app/actions/views/post` that way we could call this function when editing a post (in the case of editing a post, the  message would be only about the diff between the edited and original post) ;)

as for createPost, we could return an error if failed and keep `return {data: true}`  if it succeeds, then in the component (without awaiting, but using the Promise resolution) you call the `recentUsedEmojisInMessage` function.

Another reason to have a different action creator is to keep separation of concerns.

Reply:
@enahum currently How can I mock `createPost` to return a thenable object in jest? Currently `jest.fn()` is used. But `jest.fn()` returns `undefined`. So the test case would fail if `createPost(post, file).then()` is in  the code.

Reply:
cc @josephbaylon

Reply:
@a-c-sreedhar-reddy I believe you can use `nock` to mock the http response of `Client4.createPost`, then you can do something like,

```
        nock(Client4.getBaseRoute()).
            post('/posts').
            reply(201, {...post1, id: TestHelper.generateId()});
        post1 = await Client4.createPost(post1);
```

I believe `app/mm-redux/actions/posts.test.js` would already have examples on mocking `Client4.createPost`

However, if you're moving your functionality to redux then you just need to jest mock the redux function from your component and not mock `Client4.createPost` directly.

Let me know if I answered your question.